credentialproviderpackage/pkg/configurator/bottlerocket/bottlerocket.go

```diff
@@ -0,0 +1,172 @@
+package bottlerocket
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"credential-provider/pkg/configurator"
+	"credential-provider/pkg/constants"
+)
+
+type bottleRocket struct {
+	client  http.Client
+	baseURL string
+	config  constants.CredentialProviderConfigOptions
+}
+
+type awsCred struct {
+	Aws Aws `json:"aws"`
+}
```

Comment:
maybe Aws could be lower case? maybe if this are messages they should have unique type name prefixes to group them together